Bmx.CommandLine/IniConfiguration.cs

```diff
@@ -0,0 +1,81 @@
+﻿#nullable enable
+using System;
+using System.IO;
+using System.Security;
+using Microsoft.Extensions.Configuration;
+
+namespace Bmx.CommandLine {
+	public class IniConfiguration : IBmxConfig {
+		public string? Org { get; }
+		public string? User { get; }
+		public string? Account { get; }
+		public string? Role { get; }
+		public string? Profile { get; }
+
+		public IniConfiguration() {
+			var config = GetConfiguration();
+
+			Org = config["org"];
+			User = config["user"];
+			Account = config["account"];
+			Role = config["role"];
+			Profile = config["profile"];
+		}
+
+		private static IConfiguration GetConfiguration() {
+			// Main config is at ~/.bmx/config
+			var configLocation =
+				Path.Join( Environment.GetFolderPath( Environment.SpecialFolder.UserProfile ), ".bmx", "config" );
+
+			// If set, we recursively look up from CWD (all the way to root) for additional bmx config files (labelled as .bmx)
+			var shouldReadProjectConfig = ShouldReadProjectConfigs( configLocation );
+			string? projectConfigPath = null;
+
+			if( shouldReadProjectConfig ) {
+				projectConfigPath = GetProjectConfigPath();
+			}
+
+			var config = new ConfigurationBuilder().AddIniFile( configLocation );
+
+			if( shouldReadProjectConfig && projectConfigPath != null ) {
+				config.AddIniFile( projectConfigPath );
+			}
+
+			return config.Build();
+		}
+
+		private static bool ShouldReadProjectConfigs( string primaryConfigLocation ) {
+			var config = new ConfigurationBuilder()
+				.AddIniFile( primaryConfigLocation ).Build();
+			bool.TryParse( config["allow_project_configs"], out bool allowProjectConfigs );
```

Comment:
Decided to not use that since aside from this one boolean, the rest of the config options are consumed as simple strings.

Did strictly type the config a bit by having IniConfiguration implement IBmxConfig which lists the types it expects config to have.